manager/integration/tests/test_engine_upgrade.py

```diff
@@ -747,61 +749,261 @@ def test_engine_live_upgrade_with_intensive_data_writing(client, core_api, volum
     5. Set concurrent-automatic-engine-upgrade-per-node-limit setting to 3
-    6. In a retry loop, verify that the number of volumes who
-       is upgrading engine is always smaller or equal to 3
-    7. Wait until the upgrades are completed (volumes' engine image changed,
+    6. Wait until the upgrades are completed (volumes' engine image changed,
        replica mode change to RW for attached volumes, reference count of the
        new engine image changed, all engine and replicas' engine image changed,
        etc ...)
-    8. verify the volumes' data
-
-    Case 2: DR volume
-    1. Create a backup for vol-0. Create a DR volume from the backup
-    2. Try to upgrade the DR volume engine's image to the new engine image
-    3. Verify that the Longhorn API returns error. Upgrade fails.
-    4. Set concurrent-automatic-engine-upgrade-per-node-limit setting to 0
-    5. Try to upgrade the DR volume engine's image to the new engine image
-    6. Wait until the upgrade are completed (volumes' engine image changed,
+    7. verify the volumes' data
+    """
+    # Precondition
+    default_img, default_img_name, engine_upgrade_image, \
+        _, compatible_img_name = \
+        prepare_auto_upgrade_engine_to_default_version(client)
+
+    # Test auto upgrade to default engine in attached / detached volume
+    volume1 = client.create_volume(name="vol-1", size=str(1 * Gi),
+                                   numberOfReplicas=REPLICA_COUNT)
+    volume2 = client.create_volume(name="vol-2", size=str(1 * Gi),
+                                   numberOfReplicas=REPLICA_COUNT)
+    volume1 = common.wait_for_volume_detached(client, volume1.name)
+    volume2 = common.wait_for_volume_detached(client, volume2.name)
+
+    volume1.attach(hostId=get_self_host_id())
+    volume1 = wait_for_volume_healthy(client, volume1.name)
+    data = write_volume_random_data(volume1)
+
+    volume1.engineUpgrade(image=engine_upgrade_image)
+    volume2.engineUpgrade(image=engine_upgrade_image)
+
+    volume1 = wait_for_volume_current_image(client, volume1.name,
+                                            engine_upgrade_image)
+    volume2 = wait_for_volume_current_image(client, volume2.name,
+                                            engine_upgrade_image)
+
+    volume1 = wait_for_volume_replicas_mode(client, volume1.name, "RW")
+
+    wait_for_engine_image_ref_count(client, default_img_name, 0)
+
+    update_setting(client,
+                   "concurrent-automatic-engine-upgrade-per-node-limit",
+                   "3")
+
+    wait_for_engine_image_ref_count(client, compatible_img_name, 0)
+
+    wait_for_volume_healthy(client, volume1.name)
+    check_volume_data(volume1, data)
+
+    volume1 = client.by_id_volume(volume1.name)
+    volume2 = client.by_id_volume(volume2.name)
+    assert volume1.engineImage == default_img.image
+    assert volume2.engineImage == default_img.image
+    check_replica_engine(volume1, default_img.image)
```

Comment:
Why are we skipping the check for `volume2`?

Reply:
Hi @c3y1huang , I added volume2 replica engine check